extensions/amp-story/1.0/amp-story-form.js

```diff
@@ -0,0 +1,146 @@
+import {Action, getStoreService} from './amp-story-store-service';
+import {DrawerState} from './amp-story-draggable-drawer';
+import {LoadingSpinner} from './loading-spinner';
+import {LocalizedStringId} from '#service/localization/strings';
+import {devAssert} from '../../../src/log';
+import {escapeCssSelectorIdent} from '#core/dom/css-selectors';
+import {getLocalizationService} from './amp-story-localization-service';
+import {htmlFor} from '#core/dom/static-template';
+
+/**
+ * @enum {string}
+ */
+const FormResponseAttribute = {
+  SUBMITTING: 'submitting',
+  SUCCESS: 'submit-success',
+  ERROR: 'submit-error',
+};
+
+/**
+ * Adds AMP form actions to the action allow list.
+ * @param {!DraggableDrawer} drawerEl The draggable drawer AMP element to which
+ *     the form belongs.
+ */
+export function allowlistFormActions(drawerEl) {
+  const storeService = getStoreService(drawerEl.win);
+  storeService.dispatch(Action.ADD_TO_ACTIONS_ALLOWLIST, [
+    {tagOrTarget: 'FORM', method: 'clear'},
+    {tagOrTarget: 'FORM', method: 'submit'},
+  ]);
+}
+
+/**
+ * Add a default form attribute element for each absent response attribute.
+ * @param {!DraggableDrawer} drawerEl The draggable drawer AMP element to which
+ *     the form belongs.
+ * @param {!Element} formEl The form to which the attribute elements will be
+ *     added.
+ * @private
+ */
+export function setupResponseAttributeElements(drawerEl, formEl) {
```

Comment:
Passing the `drawerEl` here makes this function impossible to re-use in other contexts.
I think passing `win` should allow you to do all you need. Given you haven't been able to get `scrollIntoView` to work for all cases, it'd be OK to drop it.

Reply:
I updated this function so that `win` is passed, and I updated the scrolling logic so that it is instead set up from within the `amp-story-page-attachment.js`. I think we can keep the scrolling logic to help prevent our forms from appearing unresponsive.